File: frontend/src/api/notifications.ts
import { request } from './client'

export interface NotificationItem {
  id: string
  application_id: string
  kind: 'submitted' | 'resubmitted' | 'status_changed'
  title: string
  body: string
  read_at: string | null
  created_at: string
}

export interface Notifications {
  items: NotificationItem[]
  unread_count: number
}

export function getNotifications(): Promise<Notifications> {
  return request<Notifications>('/notifications')
}

export function markNotificationRead(id: string): Promise<NotificationItem> {
  return request<NotificationItem>(`/notifications/${id}/read`, { method: 'POST' })
}

export function markAllNotificationsRead(): Promise<Notifications> {
  return request<Notifications>('/notifications/read-all', { method: 'POST' })
}

File: backend/app/schemas/notifications.py
"""Notification schemas (FR-020, FR-021)."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    application_id: uuid.UUID
    kind: str
    title: str
    body: str
    read_at: datetime | None
    created_at: datetime


class NotificationsOut(BaseModel):
    items: list[NotificationOut]
    unread_count: int
